target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata

```diff
@@ -140,6 +140,11 @@ case "$FIRMWARE" in
 		caldata_valid "202f" || caldata_extract "ART" 0x1000 0x2f20
 		ath10k_patch_mac $(macaddr_add $(get_mac_binary "/sys/bus/i2c/devices/0-0050/eeprom" 0x66) +2)
 		;;
+	mikrotik,hap-ac2)
+		wlan_data="/sys/firmware/mikrotik/hard_config/wlan_data"
+		( [ -f "$wlan_data" ] && caldata_sysfsload_from_file "$wlan_data" 0x0 0x2f20 ) || \
+		( [ -d "$wlan_data" ] && caldata_sysfsload_from_file "$wlan_data/data_0" 0x0 0x2f20 )
```

Comment:
This will create subshells in addition to logic grouping. Will this have any side effects?
(Same below)

Reply:
Not sure what kind of side effect you're thinking of, besides a small memory usage increase?
This could be rewritten to use explicit conditionals instead I suppose.

Reply:
What I mean is this could be written as:
```
		if [ -f "$wlan_data" ]; then caldata_sysfsload_from_file "$wlan_data" 0x0 0x2f20
		elif [ -d "$wlan_data" ]; then caldata_sysfsload_from_file "$wlan_data/data_0" 0x0 0x2f20
		fi
```
@adschm would that be ok?

Reply:
I can also live with the subshells, I just wanted to make sure somebody thought about it at least briefly.